### PR TITLE
feat: add ts loader & checker in web builder

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/web-builder/src/plugins/tsChecker.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                typescriptPath: require.resolve('typescript'),
+              },
+              issue: {
+                include: [{ file: '**/src/**/*' }],
+                exclude: [
+                  { file: '**/*.(spec|test).ts' },
+                  { file: '**/node_modules/**/*' },
+                ],
+              },
+              ...(config.tools?.tsChecker || {}),
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/plugins/tsChecker.ts:43]: tools?.tsChecker 也要支持一下函数用法，和其他参数保持一致

File: packages/builder/web-builder/src/plugins/tsLoader.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { applyOptionsChain } from '@modern-js/utils';
+import { BuilderPlugin } from '../types';
+import { applyScriptCondition, getUseBuiltIns } from './babel';
+import { ScriptTarget, ModuleKind } from 'typescript';
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/plugins/tsLoader.ts:4]: 建议 lazy import typescript，这个包还挺大的

File: packages/builder/web-builder/src/plugins/tsLoader.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { applyOptionsChain } from '@modern-js/utils';
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/plugins/tsLoader.ts:1]: dynamic import @modern-js/utils

File: packages/builder/web-builder/src/plugins/tsLoader.ts (hunk 2 above)
Comment (sanyuan0704) [reply packages/builder/web-builder/src/plugins/tsLoader.ts:4]: 这是个类型导入，不会在运行时导入

Comment (sanyuan0704) [reply packages/builder/web-builder/src/plugins/tsLoader.ts:4]: 这个包我还是去掉吧，只拿 target 和 module，感觉没有必要引这么大一个包

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**